### PR TITLE
Deprecate Native API

### PR DESCRIPTION
**Pull request by tanwarsh**
Deprecate Python API:

1. Mark Python native API as deprecated in the doc
2. Remove Python native API code.
3. Remove github workflow realted code.
4. Remove test related to Python native API.
5. Remove Tutorials related to Python native API.

This PR will be merged after Interactive API Deprecation https://github.com/securefederatedai/openfl/pull/1125

### CODE REVIEW

File: docs/developer_guide/advanced_topics/overriding_agg_fn.rst
Comment (teoparvanov) [inline docs/developer_guide/advanced_topics/overriding_agg_fn.rst]: If we're deleting the entire `openfl.native` package, this section should be deleted too (instead of being deprecated). Indeed, what would be the point of keeping it around if there's no code backing it?

File: docs/developer_guide/advanced_topics/overriding_plan_settings.rst
Comment (teoparvanov) [inline docs/developer_guide/advanced_topics/overriding_plan_settings.rst]: Same here - remove the entire "Python API" section, instead of marking it as "deprecated".

File: docs/developer_guide/running_the_federation.notebook.rst
Comment (teoparvanov) [inline docs/developer_guide/running_the_federation.notebook.rst:7]: This isn't related to Python API, so please revert this change

File: docs/get_started/examples.rst
Comment (teoparvanov) [inline docs/get_started/examples.rst]: As mentioned above - suggesting to delete the entire "Python Native API" section

File: docs/developer_guide/utilities/splitters_data.rst
Comment (teoparvanov) [inline docs/developer_guide/utilities/splitters_data.rst]: Remove section (only leave Option 2)

File: docs/get_started/examples/python_native_pytorch_mnist.rst
Comment (teoparvanov) [inline docs/get_started/examples/python_native_pytorch_mnist.rst:1]: Shouldn't this entire file be deleted?

File: docs/source/api/openfl_native.rst
Comment (teoparvanov) [inline docs/source/api/openfl_native.rst:1]: Same here - maybe delete the entire file?

**Review by teoparvanov — COMMENTED** [review]
LGTM for a first pass (assuming deletion of the Python Native API, rather than deprecation). Could you make sure that the CI passes, so that we can have a good understanding of the entire impact of the deletion?

**Review by MasterSkepticista — APPROVED** [review]
No additional comments, looks OK.

File: docs/developer_guide/running_the_federation.notebook.rst
Comment (tanwarsh) [reply docs/developer_guide/running_the_federation.notebook.rst:7]: @teoparvanov, I think we should rewrite/delete this as the title is Aggregator based but the the tutorial is using python native API.
https://openfl.readthedocs.io/en/latest/developer_guide/running_the_federation.notebook.html

**Comment by tanwarsh** [comment]
> LGTM for a first pass (assuming deletion of the Python Native API, rather than deprecation). Could you make sure that the CI passes, so that we can have a good understanding of the entire impact of the deletion?

@teoparvanov  All CI pipelines are passing except 3 which will be removed/work once we merge https://github.com/securefederatedai/openfl/pull/1125.

**Review by rahulga1 — APPROVED** [review]

**Review by rajithkrishnegowda — APPROVED** [review]

File: docs/developer_guide/running_the_federation.notebook.rst
Comment (teoparvanov) [reply docs/developer_guide/running_the_federation.notebook.rst:7]: Then my suggestion is to delete the entire page.

**Comment by kminhta** [comment]
Overall looks good. I am in favor of entire removal of docs, workflows, tests, tutorials, etc.

Regarding impact, the entire deletion of Native API will break the [FeTS challenge](https://github.com/FeTS-AI/Challenge), which is run by our collaborators at IU and an important part of our contribution to the medical image analysis community. Ultimately, the FeTS challenge needs to move away from the Native API, there is no debate there, but we should consider maintaining functionality for another version (i.e. openfl.deprecrated.native) before entire deletion, or otherwise move it to a separate branch (this can be discussed more offline)

**Comment by MasterSkepticista** [comment]
@tanwarsh Because the tip has moved way ahead, a better option is to create patches of your changes, close this PR, and create a new branch from develop with your patches applied.

**Comment by tanwarsh** [comment]
Closing this PR due to rebase issues. Create new PR https://github.com/securefederatedai/openfl/pull/1153 for the same.

**Pull request closed without merging**
